app/controllers/overrides/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Overrides
+  class OmniauthCallbacksController <
+    DeviseTokenAuth::OmniauthCallbacksController
+    def assign_provider_attrs(user, auth_hash)
+      all_attrs = auth_hash['info'].slice(*user.attributes.keys)
+      orig_val = ActionController::Parameters.permit_all_parameters
+      ActionController::Parameters.permit_all_parameters = true
+      permitted_attrs = ActionController::Parameters.new(all_attrs)
+      permitted_attrs.permit({})
+      user.assign_attributes(permitted_attrs)
+      ActionController::Parameters.permit_all_parameters = orig_val
+      user
+    end
+
+    def get_resource_from_auth_hash
+      logger.debug auth_hash
+      # find or create user by provider and provider uid
+      @resource = resource_class.find_by(
+        uid: auth_hash['uid'],
+        provider: auth_hash['provider']
+      )
+
+      if @resource.nil?
+        @resource = User.new
+        set_resource_fields
+      end
+
+      if @resource.new_record?
```

Comment:
Are you sure this is still needed since you do the creation explicitly now inside the condition in line 23. Can't you do this inside the same condition up there?

Reply:
Done.